File: JavaScript/Algorithms/Sorting/SelectionSort.js
const selectionSort = (array) => {
    const len = array.length;
    for(let i = 0; i < len; i++) {
        let minIndex = i;
        for(let j = i+1; j < len; j++) {
            if(array[minIndex] > array[j]) {
                minIndex = j;
            }
        }
        [array[minIndex], array[i]] = [array[i], array[minIndex]];
    }

    return array;
}

console.log(selectionSort([1, 5, 4, 7, -1, 1, 0, -5, 9, 6]));

File: C#/Algorithms/Sorting/Selection Sort.cs
using System;

namespace Sorting
{
    internal class Program
    {
        public static void Main(string[] args)
        {
            int[] array = { -4, 5, 9, -7, 1, 2, 1, -16, 16, 18, 4 };
            selectionSort(ref array);
            displayArray(ref array);
        }

        public static void selectionSort(ref int[] array)
        {
            int length = array.Length;
            for (int i = 0; i < length; i++)
            {
                
                int minIndex = i;
                for (int j = i+1; j < length; j++)
                {
                    if (array[j] < array[minIndex])
                    {
                        minIndex = j;
                    }
                }

                (array[minIndex], array[i]) = (array[i], array[minIndex]);
            }

        }
        public static void displayArray(ref int[] array)
        {
            int length = array.Length;
            for (int i = 0; i < length; i++)
            {
                Console.Write($"{array[i]}  ");
            }
            Console.WriteLine();
        }
    }
}